### PR TITLE
Add support for window, linux and macos folders

### LINKED ISSUE #650: [Request] Specific folder icons
### We are keeping here a list of requested folder icons.
We are going to update this list as we go along adding icons.

Before requesting a folder icon make sure to check out the already [supported folder list](https://github.com/vscode-icons/vscode-icons/wiki/ListOfFolders).

**Any new requests SHOULD be submitted via comments on this issue and we'll update the list accordingly.**

- [ ] arangodb
- [ ] cmake
- [ ] console, cli, commandline
- [ ] controllers
- [ ] data
- [ ] enum
- [ ] environments
- [ ] gulp
- [ ] graphql
- [ ] grunt
- [ ] minikube
- [ ] `__mocks__`, mocks
- [ ] polyfills
- [ ] resources
- [ ] serializers
- [ ] services
- [ ] shared, common, core

### Debatable
`Debatable` are folder names that are debatable on whether we should provide a specific icon.

- [ ] vendor

### PR DESCRIPTION
**Pull request by KingDarBoja**
_**Fixes #650**_

**Changes proposed:**

- [x] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

Added folder icons as suggested by the issue but wasn't sure which color use for each folder, so feel free to change it. 👍

### CODE REVIEW

**Comment by JimiC** [comment]
OK. Let's explain how we chose the color of the folder icon.
The main principal follows this simple concept: Chose a color from the overlaying icon that will have the greatest contrast with the overlaying icon when applied in the background.

Once you picked the color, inspect it in `Illustrator`'s color picker, observe the `H` value and apply the same value in the `H` field of the background icon.
The `default_folder` has a starting value of `35` and the `default_folder_opened` has a starting value of `36`. The final value of `H` for `default_folder` has to be the same as the `H` value of the color you picked from the overlaying icon and the final value of `H` for `default_folder_opened` has to be +1 of the `H` value of the color you picked from the overlaying icon.

For example, for the `linux` icons I choose the `yellow` color because it contrasts with the `black`. The `yellow` color has an `H` value of 42. So for the `default_folder`, the `H` value has to become 42 and for the `default_folder_opened` it has to become 43.

Note here that entering the exact number (42) in the `H`field in `Illustrator` does not take that number. In many cases, you have to enter 2 or 3 numbers higher in order for `Illustrator` to take the correct number. due to same adjusting calculations it does.

File: icons/folder_type_windows.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><title>folder_type_windows</title><path d="M27.9,6H18.1l-2,4H5V27H30V6Zm.1,4H20.2l1-2H28Z" style="fill:#55a1bf"/><path d="M10,12.974l8.582-1.166v8.253L10,20.109Zm8.577,8.037.008,8.261L10.006,28.1v-7.14Zm1.041-9.355L31,10v9.956l-11.379.089ZM31,21.089V31L19.618,29.4,19.6,21.07Z" style="fill:#00adef"/></svg>
```
Comment (robertohuertasm) [inline icons/folder_type_windows.svg]: `Windows 10` blue color is a [little bit darker](https://commons.wikimedia.org/wiki/File:Windows_10_Logo.svg). We're using the `Windows 8` blue here.

Comment (JimiC) [reply icons/folder_type_windows.svg]: Adjusting....

**Comment by KingDarBoja** [comment]
@JimiC  We could include those steps into the wiki so contributors can get some insight while making folder icons (like me) 😆

**Comment by JimiC** [comment]
I'm too lazy with docs.

**Comment by KingDarBoja** [comment]
> I'm too lazy with docs.

Just added those steps into the wiki `contributing icons` section.

**Comment by robertohuertasm** [comment]
Thanks for that @KingDarBoja! I really appreciate it! 

On the other hand, I wasn't aware that our wiki was editable by everyone! I've just changed that to limit the edition to members with push privileges. I wouldn't want that some random internet troll could change or delete some of our documentation.

@KingDarBoja, you're getting close to get the offer of becoming a full member/maintainer! You're helping us a lot with this project and we are really grateful for that! ❤️

**Comment by JimiC** [comment]
@robertohuertasm I have deliberately left the wiki open so that the community can help us improve the docs. In case of trolling we can always revert any change as the wiki is also a repo. I'm sure that in case of troll the community will notify us asap.

**Pull request merged**